### ISSUE TITLE
Does PointGroup uses rgb?

### ISSUE THREAD

**Issue by juanmed** [nicolas-chaulet/torch-points3d#652]
Hello,

Thanks again for this amazing repo. I was wondering if the Pointgroup implementation in this repo is using the rgb data? The PointGroup paper mentions that the input are points made of a 3d coordinate (x,y,z) and color (r,g,b). However, looking at the ```PanopticLabels``` class and the ```__REQUIRED_DATA``` member in ```pointgroup.py``` it seems that pointgroup is not using the color of each point as input. The ```PointGroup``` model uses the ```Minkowski```as backbone but I could not find any reference to the color in Minkowski. Please correct me if I am wrong.

https://github.com/nicolas-chaulet/torch-points3d/blob/98eed17c6bf4d38c33325d4f29357076c35a3f07/torch_points3d/models/panoptic/structures.py#L52-L58

https://github.com/nicolas-chaulet/torch-points3d/blob/98eed17c6bf4d38c33325d4f29357076c35a3f07/torch_points3d/models/panoptic/pointgroup.py#L19-L21

If PointGroup's implementation in this repo is not using color as in the paper, would you provide some indication on how to include the color  in PointGroup? 

Thanks

**Comment by juanmed** [comment]
After studying over the source code, it seems that on the ```datatasets/segmentation/scannet.py``` file there is loading of the rgb features at: 

https://github.com/nicolas-chaulet/torch-points3d/blob/4822798e5a256dcb627aae657c2562a7c5b6887f/torch_points3d/datasets/segmentation/scannet.py#L623

Then, in ```models/panoptic/pointgroup.py``` the ```input`` data is read at:

https://github.com/nicolas-chaulet/torch-points3d/blob/98eed17c6bf4d38c33325d4f29357076c35a3f07/torch_points3d/models/panoptic/pointgroup.py#L65

This ```input``` is passed to the backbone, which, when initialized here: 

https://github.com/nicolas-chaulet/torch-points3d/blob/98eed17c6bf4d38c33325d4f29357076c35a3f07/torch_points3d/models/panoptic/pointgroup.py#L28

is passed the ```feature_dimension```. I assume that the feature dimension is 6 (x,y,z,r,g,b). This all makes sense but when I searched for the definition of  ```feature_dimension``` for pointgroup, I could not find it anywhere. In the pretrained API ```torch-points3d/applications/pretrained_apy.py``` there are definitions for other network-dataset pairs but not for pointgroup-scannet:

https://github.com/nicolas-chaulet/torch-points3d/blob/98eed17c6bf4d38c33325d4f29357076c35a3f07/torch_points3d/applications/pretrained_api.py#L72

I imagine this should be set manually. I will try to follow this for my custom dataset and hopefully it will work when setting ```dataset.feature_dimension = 6```

**Comment by juanmed** [comment]
I am now able to train pointgroup with my custom dataset. I made practically no changes to pointgroup. However I am still unsure whether or not the backbone based on Minkowski engine is using only the ```pos``` or if it is using  ```rgb``` or any other fields of each ```Data```. Reading the code in Minkowski engine, it has no references to the rgb values.

In the following line:
https://github.com/nicolas-chaulet/torch-points3d/blob/4822798e5a256dcb627aae657c2562a7c5b6887f/torch_points3d/applications/minkowski.py#L122

the input that will be passed through ```forward``` is obtained from ```data.x``` which is ```None``` (following scannet dataset) and the coords (which are obtained after doing gridTransformation from ```data.pos```).  Since the backbone is of type ```Unet``` then in forward method the fields used are ```data.F```, ```self.xyz = data.pos``` and ```data.C[:,0]```

https://github.com/nicolas-chaulet/torch-points3d/blob/4822798e5a256dcb627aae657c2562a7c5b6887f/torch_points3d/applications/minkowski.py#L193

So I am still lost on whether or not the backbone uses ```pos``` and ```rgb``` only, or only ```pos``` or if it is using all features in the ```Data``` object. 

Any comments are welcome.

**Comment by juanmed** [comment]
It seems I completely missed the ```AddFeatsByKeys``` transform in the config file for the panoptic datasets. This adds the ```rgb``` feature to ```data.x``` and with that all previous makes sense.  Before, when training, the loss values never decreased. I will try now by adding this transform and hopefully some learning will happen. 

https://github.com/nicolas-chaulet/torch-points3d/blob/87c58c60c4a73449665e6589ee7bb58ea2fe9062/conf/data/panoptic/scannet-sparse.yaml#L43-L47

**Comment by juanmed** [comment]
I am unable to train pointgroup but I think that is an unrelated issue. I will close this.
